src/ImageSharp/Formats/Png/Filters/AverageFilter.cs

```diff
@@ -22,18 +22,26 @@ internal static unsafe class AverageFilter
         /// <param name="bytesPerScanline">The number of bytes per scanline</param>
         /// <param name="bytesPerPixel">The bytes per pixel.</param>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static void Decode(byte[] scanline, byte[] previousScanline, int bytesPerScanline, int bytesPerPixel)
+        public static void Decode(BufferSpan<byte> scanline, BufferSpan<byte> previousScanline, int bytesPerScanline, int bytesPerPixel)
         {
+            ref byte scanPointer = ref scanline.DangerousGetPinnableReference();
```

Comment:
Maybe we need a common naming rule for these variables. 
I have to confess that I'm also mixing stuff: sometimes I name them `blahFooBarPointer` on other places I use `blahFooBarBase`. This is confusing.